discovery/config/src/main/java/org.apache.aries.rsa.discovery.config/Activator.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.aries.rsa.discovery.config;
+
+import org.osgi.framework.*;
+import org.osgi.service.cm.ManagedServiceFactory;
+import org.osgi.service.remoteserviceadmin.EndpointListener;
+import org.osgi.util.tracker.ServiceTracker;
+
+import java.util.Hashtable;
+
+@SuppressWarnings("deprecation")
+public class Activator implements BundleActivator {
+    private static final String FACTORY_PID = "org.apache.aries.rsa.discovery.config";
+
+    private ServiceTracker<EndpointListener, EndpointListener> listenerTracker;
+    private ConfigDiscovery configDiscovery;
+    private ServiceRegistration<ManagedServiceFactory> registration;
+
+    public void start(BundleContext context) {
+        configDiscovery = new ConfigDiscovery();
+        listenerTracker = new EPListenerTracker(context, configDiscovery);
+        listenerTracker.open();
+        registration = context.registerService(ManagedServiceFactory.class, configDiscovery, new Hashtable<String, Object>() {{
```

Comment:
Better not use the static put style. It creates a anon subclass and spits a serial warning.